x-pack/plugins/stack_connectors/server/connector_types/openai/openai.ts

```diff
@@ -51,6 +64,23 @@ export class OpenAIConnector extends SubActionConnector<Config, Secrets> {
     this.provider = this.config.apiProvider;
     this.key = this.secrets.apiKey;
 
+    this.openAI =
+      this.config.apiProvider === OpenAiProviderType.AzureAi
+        ? new OpenAI({
+            apiKey: this.secrets.apiKey,
+            baseURL: this.config.apiUrl,
+            defaultQuery: { 'api-version': getAzureApiVersionParameter(this.config.apiUrl) },
+            defaultHeaders: { 'api-key': this.secrets.apiKey },
```

Comment:
another thing is that you'll want to pass the headers that are being added in https://github.com/Portkey-AI/gateway?

Reply:
ok will do!

Reply:
Your PR needs to merge first, but its ready to go here: https://github.com/elastic/kibana/pull/174126/commits/dd6bfa913a182f79442ec9d1fe0e08c3b4c491ec